cli/config.go

```diff
@@ -202,6 +203,28 @@ func (h hypervisor) defaultVCPUs() uint32 {
 	return uint32(h.DefaultVCPUs)
 }
 
+func (h hypervisor) defaultMaxVCPUs() uint32 {
+	numcpus := goruntime.NumCPU()
+	maxvcpus := vc.MaxQemuVCPUs()
+
+	if h.DefaultMaxVCPUs <= 0 {
+		// Don't exceed the maximum number of vCPUs supported by QEMU
+		if numcpus > int(maxvcpus) {
+			return maxvcpus
+		}
+		return uint32(numcpus)
+	}
+
+	if h.DefaultMaxVCPUs > int32(numcpus) {
```

Comment:
do we need this comparison?

Reply:
yes, in case of user value is greater than the actual number of physical CPUs, we need to check this values doesn't exceed the maximum number of vCPUs supported by QEMU/KVM.
Adding a comment in the code.
Thanks

Reply:
ohhh wait.. docker and kubernetes don't allow you to create containers with a number of CPUs greater than the actual number of physical CPUs, hence we don't need this condition, thanks

```
$ docker run --cpus 10 -ti centos bash
docker: Error response from daemon: Range of CPUs is from 0.01 to 8.00, as there are only 8 CPUs available.
```